src/perl/irc/Irc.xs

```diff
@@ -34,10 +37,14 @@ static void perl_irc_server_fill_hash(HV *hv, IRC_SERVER_REC *server)
 	(void) hv_store(hv, "cap_complete", 12, newSViv(server->cap_complete), 0);
 	(void) hv_store(hv, "sasl_success", 12, newSViv(server->sasl_success), 0);
 
-	av = newAV();
-	for (tmp = server->cap_supported; tmp != NULL; tmp = tmp->next)
-		av_push(av, new_pv(tmp->data));
-	(void) hv_store(hv, "cap_supported", 13, newRV_noinc((SV*)av), 0);
+	hv_ = newHV();
+	g_hash_table_iter_init(&iter, server->cap_supported);
+	while (g_hash_table_iter_next(&iter, &key_, &val_)) {
+		char *key = (char *)key_;
+		char *val = (char *)val_;
+		hv_store(hv_, key, strlen(key), new_pv(val), 0);
+	}
+	(void) hv_store(hv, "cap_supported", 13, newRV_noinc((SV*)hv_), 0);
```

Comment:
So this is a perl api change, right? Does perl handle this decently enough, falling back to keys only if it is accessed like an array instead of a hash? @ailin-nemui

Reply:
no that would not work..

Reply:
(but I guess we can break the API)

Reply:
My (wild) guess is that we have no users of this API heh
I'd avoid such a breaking change in a non-point release though, got a better idea that's backward compatible?

Reply:
well, I can't think of any way other than adding another key, say cap_supported_hash or another method, since we initially put it it all in an array. but that seems silly imo